sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -663,14 +663,9 @@ class ResolveSessionCatalog(
   object TempViewOrV1Table {
     def unapply(nameParts: Seq[String]): Option[Seq[String]] = nameParts match {
       case _ if isTempView(nameParts) => Some(nameParts)
-      case SessionCatalogAndTable(_, tbl) =>
-        if (nameParts.head == CatalogManager.SESSION_CATALOG_NAME && tbl.length == 1) {
-          // For name parts like `spark_catalog.t`, we need to fill in the default database so
-          // that the caller side won't treat it as a temp view.
-          Some(Seq(catalogManager.v1SessionCatalog.getCurrentDatabase, tbl.head))
-        } else {
-          Some(tbl)
-        }
+      case Seq(CatalogManager.SESSION_CATALOG_NAME, viewName @ _*) if isTempView(viewName) =>
+        throw new AnalysisException("Temp view name cannot contain catalog name")
```

Comment:
@cloud-fan Is this in line with what you had in mine?

Reply:
I think we should fail with no such table/database if when see something like `spark_catalog.tempViewName`

Reply:
Ok, we can just return `Some(nameParts)` here if we want that behavior.